app/scrapers/__init__.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import print_function
+import os, json, sys
+from google import Google
+from duckduckgo import Duckduckgo
+from bing import Bing
+from yahoo import Yahoo
+import requests
+from bs4 import BeautifulSoup
+scrapers = {
+        'google':Google(),
+        'bing':Bing(),
+        'yahoo':Yahoo(),
+        'duckduckgo':Duckduckgo(),
+    }
+
+def read_in():
+    lines = sys.stdin.readlines()
+    return json.loads(lines[0])
+
+
+def small_test():
+    assert type(scrapers.google.results_search('fossasia')) is list
+
+
+def feedgen(query, engine):
+    if engine == 'g':
```

Comment:
was thinking instead of so many if elif blocks, wouldn't it be more efficient to set the dictionary with engine code, that way you can directly run `scrapers[engine].resets_search(query)`